icicle-compiler/src/Icicle/Command/Compile.hs

```diff
@@ -150,3 +158,18 @@ icicleCompile compile = do
   liftIO $ writeSeaDictionary (compileOutputDictionary compile) sea
 
   finishAll
+
+----------------
+-- Check Only --
+----------------
+
+icicleCheck :: Check -> EitherT CompileError IO ()
+icicleCheck check = do
+  dictionary <- loadDictionary (checkInputDictionary check)
+  let
+    summary
+      = prettyDictionarySummary dictionary
+    render
+      = Text.pack (show (Pretty.pretty summary))
```

Comment:
Wat? Does that really need `show` *and* `Text.pack`?

Reply:
Not really, I could just use `print` on the output of Pretty.pretty.

I did this to keep in line with the error message rendering higher up in the file.

Reply:
I'll change it to putDoc.

Reply:
What is the type of `Pretty.pretty`? We usually try to avoid `Show` instances for anything other than printing stuff in GHCi and because `Show` instances are required in quickcheck properties.